### PR TITLE
Join HMIS Enrollment-related models on PersonalID+EnrollmentID

### PR DESCRIPTION
**Pull request by gigxz**
## Description

* Update Enrollment-related records to only join with Enrollments where the PersonalID matches, to match warehouse behavior.
* Unrelated bug fix to how household assessments are grouped

## Type of change
- [x] Bug fix
- [ ] New feature (adds functionality)
- [ ] Code clean-up / housekeeping
- [ ] Dependency update

## Checklist before requesting review
- [ ] I have performed a self-review of my code
- [ ] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [ ] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [ ] My code follows the style guidelines of this project (rubocop)
- [ ] I have updated the documentation (or not applicable)
- [ ] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: drivers/hmis/app/models/hmis/hud/custom_assessment.rb [hunk 1]
```diff
@@ -163,7 +163,9 @@ def self.group_household_assessments(household_enrollments:, assessment_role:, t
     raise HmisErrors::ApiError, 'Assessment not in household' if source_assessment.present? && !household_enrollments.pluck(:enrollment_id).include?(source_assessment.enrollment_id)
 
     household_assessments = Hmis::Hud::CustomAssessment.with_role(assessment_role.to_sym).
-      where(enrollment_id: household_enrollments.pluck(:enrollment_id))
+      where(enrollment_id: household_enrollments.pluck(:enrollment_id), data_source_id: household_enrollments.pluck(:data_source_id))
+
+    threshold_in_days = threshold.to_i / 86400 # to_i always returns seconds
```
Comment (gigxz) [inline drivers/hmis/app/models/hmis/hud/custom_assessment.rb:168]: unrelated bug fix! was surfaced when updating tests..

File: drivers/hmis/app/models/hmis/hud/enrollment_coc.rb
Comment (eanders) [inline drivers/hmis/app/models/hmis/hud/enrollment_coc.rb:1]: Do we need this class any more?

**Review by eanders — APPROVED** [review]

File: drivers/hmis/app/models/hmis/hud/enrollment_coc.rb
Comment (gigxz) [reply drivers/hmis/app/models/hmis/hud/enrollment_coc.rb:1]: no, we don't - will remove in a separate PR though
